rust_src/src/frames.rs

```diff
@@ -727,6 +727,33 @@ pub fn make_frame_invisible(frame: LispFrameLiveOrSelected, force: bool) {
     }
 }
 
+#[lisp_fn(min = "0")]
+pub fn last_nonminibuffer_frame() -> LispObject {
+    let frame = get_last_nonminibuffer_frame();
+
+    if frame.is_null() {
```

Comment:
I wonder if this exposes a bug/failure in the current code. The `impl` for LispObject could check for NULL and return Qnil. Otherwise do we silently transform the null into a bogus pointer?

Reply:
Yes. I assume you are right and current `from` implementation is wrong.

In theory the function, which I implemented should be a one-liner `get_last_nonminibuffer_frame().into()`, but since current `from` implementation doesn't work this way (idk why, maybe it's a bug), I couldn't do it like that.

I think of making a new PR, where I fix the `from` method and if it's gonna be merged I will polish this one. What do you think?

Reply:
> I think of making a new PR, where I fix the from method and if it's gonna be merged I will polish this one. What do you think?

this is a solid plan. Maybe there are reasons for the current implementation. Maybe not. If nothing else, we can improve the comments in the code.

Reply:
TL;DR I propose altering my API instead and change `the last_non_minibuffer_frame` from `LispFrameRef` to `Option<LispFrameRef>`

The extern functions, which are exposed to C indeed can pass null reference hidden under LispFrameRef (e.g. frame.c line: 1838), but our Rust code should always point to the non-null frame (can we make a wrapper around this and make our Rust function receive Option<LispFrameRef>?)
i.e. in the setter, which is called from C, I will make an extra check, in order to check for nil
---
I looked at the code one more time and I think the reason why this was done this way is the assumption that if we have a `LispFrameRef` instance, then we assume that the ref is always pointing to a frame, meaning, it can not contain a pointer to `null`

According to types though, it can be point to null, since `LispFrameRef`is a type alias of `ExternalPtr<Lisp_Frame>` and `ExternalPtr` has an `is_null` method

But by checking the API, like for instance `LispObject::as_frame` returns an `Option<LispFrameRef>` and not a `LispFrameRef` which then points to null.

Reply:
Yeah, this is something none of us have been happy with. We should refactor the handling of LispFrameRef. That isn't a requirement for this PR though.

My recollection is there are edge cases where NULL is in fact possible where it acts like a None value. LispFrameRefs will be C pointers until we fully port them over to Rust, which will be a while.